### PR TITLE
Posix: fix copyright

### PR DESCRIPTION
**Pull request by unspecd**
This fixes compilation error when `configENABLE_BACKWARD_COMPATIBILITY` is 0.

### CODE REVIEW

**Comment by unspecd** [comment]
Should I squash everything into 1 commit?

**Comment by dachalco** [comment]
@unspecd 

Thanks for bringing this up. Let me check with some colleagues. Iff I recall correctly, this deficiency remained present by design.

**Comment by dachalco** [comment]
@unspecd 

Is there a particular reason for removing these macros?

The preprocessor can do the same replacement with:
```
#define pdTASK_CODE TaskFunction_t
```

Also the copyright line has been modified and I'm not sure whether the copyright line change is appropriate for `port.c`

**Comment by unspecd** [comment]
> This removes the convenient ability to redefine the task function signature in Thread_t via pdTASK_CODE

Oh, I didn't take this into account.

**Comment by unspecd** [comment]
> port.c copyright should remain unchanged 

OK, changed to "Cambridge Consultants Ltd".

File: portable/ThirdParty/GCC/Posix/portmacro.h [hunk 1]
```diff
@@ -70,7 +69,7 @@ typedef unsigned long TickType_t;
 #define portSTACK_GROWTH			( -1 )
 #define portHAS_STACK_OVERFLOW_CHECKING	( 1 )
 #define portTICK_PERIOD_MS			( ( TickType_t ) 1000 / configTICK_RATE_HZ )
-#define portTICK_RATE_MICROSECONDS	( ( portTickType ) 1000000 / configTICK_RATE_HZ )
+#define portTICK_RATE_MICROSECONDS	( ( TickType_t ) 1000000 / configTICK_RATE_HZ )
```
Comment (dachalco) [inline portable/ThirdParty/GCC/Posix/portmacro.h:72]: What are the implications of removing `portTickType` here?

`TickType_t` is limited to 16 or 32 bits and one utility of having `portTickType` here is that 64-bit posix targets can define `portTickType` as 64-bit instead, allowing finer granularity.


I'm not very familiar with how our Posix handles/quantifies tick cycles.
However I can say that

Comment (dachalco) [reply portable/ThirdParty/GCC/Posix/portmacro.h:72]: Is there specific issue that removing `portTickType` here solves? Why the removal?

Comment (unspecd) [reply portable/ThirdParty/GCC/Posix/portmacro.h:72]: There are no specific issues, I just wanted to fix a compilation error.

**Review by paulbartell — APPROVED** [review]

**Review by dan4thewin — APPROVED** [review]

**Pull request merged**